src/main/java/PasswordGenerator.java

```diff
@@ -1,8 +1,60 @@
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
 /*
  * This Java source file was generated by the Gradle 'init' task.
  */
-public class PasswordGenerator {
-    public String generatePassword() {
-        return "a+1";
+
+class PasswordGenerator extends CharacterGenerator {
+
+    String generatePassword(int passwordLength, int numbers, int specialChars) {
+
+        StringBuilder passwordStringBuilder = new StringBuilder();
+        List<Character> passwordArray = new ArrayList<>();
+
+        int charactersRemaining = passwordLength - (numbers + specialChars);
+
+        for (int i = 0; i < numbers; i++) {
```

Comment:
I think we could go a step further and push the for loop into the `RandomIndexGenerator` method, saving a bit more repetition.